exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogFormatConfig.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.log;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Objects;
+import org.apache.drill.common.logical.FormatPluginConfig;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@JsonTypeName("logRegex")
+public class LogFormatConfig implements FormatPluginConfig {
+
+  public String regex;
+  public String extension;
+  public int maxErrors = 10;
+  public List<LogFormatField> schema;
+
+  public String getRegex() {
+    return regex;
+  }
+
+  public String getExtension() {
+    return extension;
+  }
+
+  public int getMaxErrors() {
+    return maxErrors;
+  }
+
+  public List<LogFormatField> getSchema() { return schema; }
+
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+    LogFormatConfig other = (LogFormatConfig) obj;
+    return Objects.equal(regex, other.regex) &&
+        Objects.equal(maxErrors, other.maxErrors) &&
+        Objects.equal(schema, other.schema) &&
+        Objects.equal(extension, other.extension);
+  }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(new Object[]{regex, maxErrors, schema, extension});
+  }
+
+  @JsonIgnore
+  public List<String> getFieldNames(){
+    List<String> result = new ArrayList<String>();
+    if(this.schema == null ){
+      return result;
+    }
+
+    for( LogFormatField field: this.schema ){
+      result.add( field.getFieldName() );
+    }
+    return result;
+  }
+
+  @JsonIgnore
+  public String getDataType(int fieldIndex) {
+    LogFormatField f = this.schema.get(fieldIndex);
+    return f.fieldType.toUpperCase();
+  }
+
+  @JsonIgnore
+  public String getDateFormat( int patternIndex ){
+    LogFormatField f = this.schema.get(patternIndex);
+    return f.format;
+  }
+
+}
```

Comment:
General comment: this reader has gotten robust enough that it probably needs a bit of explanation. Can we include a `README.md` file (or `package-info.java`, if you prefer JavaDoc) to explain how a user should configure and use this reader?

Maybe document the supported column types, the syntax of the formats (point to the proper Java classes), and the existence of your special columns.